pkg/kubectl/cmd/expose.go

```diff
@@ -182,6 +182,17 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 			params["ports"] = strings.Join(ports, ",")
 		}
 	}
+
+	// Always try to derive protocols from the exposed object, as
+	// user may specify UDP protocol
+	protocolsMap, err := f.ProtocolsForObject(inputObject)
```

Comment:
Please follow the pattern used above:
```go
if protocol, found := params["protocol"]; found {
   // your code
}
```
This assures that only generated objects that need to use protocol will do.